sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/LogicalPlan.scala

```diff
@@ -205,8 +200,10 @@ trait UnaryNode extends LogicalPlan with UnaryLike[LogicalPlan] {
  */
 trait BinaryNode extends LogicalPlan with BinaryLike[LogicalPlan]
 
-abstract class OrderPreservingUnaryNode extends UnaryNode {
-  override final def outputOrdering: Seq[SortOrder] = child.outputOrdering
+trait OrderPreservingUnaryNode extends UnaryNode
+  with AliasAwareQueryOutputOrdering[LogicalPlan] {
+  override protected def outputExpressions: Seq[NamedExpression] = child.output
```

Comment:
this means we don't do any alias replacement?

Reply:
no it just specifies it's outputExpressions to `AliasAwareOutputExpression` so that `AliasAwareOutputExpression` can build alias map

Reply:
but `child.output` has no alias at all, right?

Reply:
`Project` has overridden it..
```
override protected def outputExpressions: Seq[NamedExpression] = projectList
```

Reply:
which subclass uses the default implementation?

Reply:
`Filter`/`Limit` etc..